libshaderc/include/shaderc.h

```diff
@@ -141,6 +141,55 @@ void shaderc_compile_options_set_disassembly_mode(
 void shaderc_compile_options_set_forced_version_profile(
     shaderc_compile_options_t options, int version, shaderc_profile profile);
 
+// When including a file, libshaderc initiates a query to get the full
+// path and the content of a file. The query is initiated through calling a
```

Comment:
I normally call these "callbacks". So you could simplify the wording.
You could simplify the first two sentences with:


```
To support file inclusion, libshaderc invokes a callback into its client to resolve the full path and content of the included file.   
```
Later phrases of 'client-side implemented method' can be replaced with 'client callback'.